File: MedicApp/Models/WorkItem.cs
namespace MedicApp.Models
{
    public class WorkItem
    {
        public Guid Id { get; set; }
        public String Name { get; set; }
        public int UsedInstances { get; set; } 
        public bool IsDeleted { get; set; }
        public DateTime Creation_TimeStamp { get; set; }
        public WorkItem()
        {
            Id = Guid.NewGuid();
            IsDeleted = false;
            Creation_TimeStamp = DateTime.Now;
        }

        public class WorkItemInfo
        {
            public String Name { get; set; }
            public int UsedInstances { get; set; }
        }
    }

}

File: Client/src/app/model/WorkItem.ts
import { WorkItemType } from "./WorkItemType";

export interface WorkItemInterface {
  id?: string;
  name?: WorkItemType;
  usedInstances?: number;
  isDeleted?: boolean;
  isChecked?: boolean;
}

export class WorkItem implements WorkItemInterface {
  id?: string;
  name?: WorkItemType;
  usedInstances?: number;
  isDeleted?: boolean;
  isChecked?: boolean;

  constructor(obj: WorkItemInterface) {
    this.id = obj.id;
    this.name = obj.name;
    this.usedInstances = obj.usedInstances;
    this.isDeleted = obj.isDeleted;
    this.isChecked = obj.isChecked;
  }
}
